core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -3607,6 +3607,25 @@ public int size() {
     return relBuilder.build();
   }
 
+  /** Explain filtering condition and projections for Calc. */
+  public static Pair<RexNode, List<RexNode>> explainCalc(Calc calc) {
```

Comment:
Could you please elaborate on what does `explainCalc` mean?

Reply:
Maybe put a simple (non-trivial) example in the javadoc. Expressions, common sub-expressions, filter, some output fields based on constants, some input fields pruned from the output.

Reply:
Sure, I'll update